pkg/mic/mic.go

```diff
@@ -99,7 +112,58 @@ func NewMICClient(cloudconfig string, config *rest.Config, isNamespaced bool, sy
 		NodeClient:        &NodeClient{informer.Core().V1().Nodes()},
 		IsNamespaced:      isNamespaced,
 		syncRetryInterval: syncRetryInterval,
-	}, nil
+	}
+	leaderElector, err := c.NewLeaderElector(clientSet, recorder, leaderElectionConfig)
+	if err != nil {
+		return nil, err
+	}
+	c.leaderElector = leaderElector
+
+	return c, nil
+}
+
+// Run - Initiates the leader election run call to find if its leader and run it
+func (c *Client) Run() {
+	glog.Infof("MIC Leader election initiated")
+	c.leaderElector.Run()
+}
+
+// NewLeaderElector - does the required leader election initialization
+func (c *Client) NewLeaderElector(clientSet *kubernetes.Clientset, recorder record.EventRecorder, leaderElectionConfig *LeaderElectionConfig) (leaderElector *leaderelection.LeaderElector, err error) {
+	c.LeaderElectionConfig = leaderElectionConfig
+	resourceLock, err := resourcelock.New(resourcelock.EndpointsResourceLock,
+		c.Namespace,
+		c.ID,
+		clientSet.CoreV1(),
+		resourcelock.ResourceLockConfig{
+			Identity:      c.ResourceName,
+			EventRecorder: recorder})
+	if err != nil {
+		glog.Errorf("Resource lock creation for leadeer election failed with error : %v", err)
+		return nil, err
+	}
+	config := leaderelection.LeaderElectionConfig{
+		LeaseDuration: c.Duration,
+		RenewDeadline: c.Duration / 2,
+		RetryPeriod:   c.Duration / 4,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(exit <-chan struct{}) {
+				c.Start(exit)
+			},
+			OnStoppedLeading: func() {
+				glog.Errorf("Lost Leader Lease")
+				glog.Flush()
+				os.Exit(1)
```

Comment:
nit the case of log line consistent. don't use title case, and generally avoid caps all together.

Reply:
Changed. Only first one is caps now.